### PR TITLE
Fix bounding box visualization tool to draw boxes inside of the box when OOB

### PR DESCRIPTION
**Pull request by LukeWood**
Also adds confidence scores to the plotting utility

Some edge case testing:

Normal font size:
Figure_2

Massive font size:

Figure_1

### CODE REVIEW

File: examples/visualization/plot_bounding_box_gallery.py [hunk 1]
```diff
@@ -80,30 +84,29 @@ def unpackage_tfds_inputs(inputs):
     images,
     value_range=(0, 255),
     bounding_box_format="xywh",
-    y_true=boxes,
-    scale=3,
+    y_pred=boxes,
```
Comment (ianstenbit) [inline examples/visualization/plot_bounding_box_gallery.py:87]: why not `y_true` here anymore?

Comment (LukeWood) [reply examples/visualization/plot_bounding_box_gallery.py:87]: I was just trying to show off the confidence scores in a way that would actually be used/logical

**Comment by LukeWood** [comment]
@ianstenbit reverted!

File: keras_cv/visualization/draw_bounding_boxes.py [hunk 2]
```diff
@@ -117,11 +128,26 @@ def draw_bounding_boxes(
...
                     cv2.FONT_HERSHEY_SIMPLEX,
                     font_scale,
                     color,
                     text_thickness,
                 )
         result.append(image)
     return np.array(result).astype(int)
+
+
+def _find_text_location(x, y, font_scale, line_thickness, outline_factor):
```
Comment (ianstenbit) [inline keras_cv/visualization/draw_bounding_boxes.py:141]: Just to sanity check -- have you tested this on a few potential edge cases? I can't totally parse the logic, but the demos look good.

**Review by ianstenbit — APPROVED** [review]
lgtm -- one question below

File: keras_cv/visualization/draw_bounding_boxes.py (hunk 2 above)
Comment (LukeWood) [reply keras_cv/visualization/draw_bounding_boxes.py:141]: I have, I tried fiddling with thicknesses and font scales and outline factors quite a bit.  I've failed to find a failure case, but if there is one we can submit new cases to handle it.

**Comment by jbischof** [comment]
Nice, thank you @LukeWood!!!

**Pull request merged**